### ISSUE TITLE
certain data fields are not showing up in the data standards page

### ISSUE THREAD

**Issue by inodb** [ncihtan/htan-portal#295]
- Conditional data fields are not showing up. E.g. `Dead `,  `Days_to_Death`, `Age_at_Index`, and `Cause_of_Death_Source` are not showing up in https://data.humantumoratlas.org/standard/clinical. They are conditional fields that only show up if other fields are filled in
- `Ischemic_Time` is also not showing up

**Comment by onursumer** [comment]
We just realized that the data schema entities only list required dependencies in our json file: https://raw.githubusercontent.com/ncihtan/schematic/main/data/schema_org_schemas/HTAN.jsonld

Conditional dependencies are not listed within an entity. The only way to retrieve them is to traverse the entire JSON and map by the `rdfs:subClassOf` field. 

@milen-sage Do you think does it make sense to update the JSON and add a field for conditional dependencies similar to required dependencies?

**Comment by milen-sage** [comment]
hi @onursumer, I see what you mean - currently fetching conditional dependencies requires recursive traversal of the schema graph. We do that on our end anyway when we generate validation schemas and manifests. We are working on a set of schema REST API endpoints, one of which would let you fetch these dependencies. I'll either point you to the endpoint or change our schema generation to include these conditional attributes directly - whichever happens first over the next weeks.

**Comment by inodb** [comment]
@milen-sage any update on this one?

**Comment by inodb** [comment]
@milen-sage @elv-sb this was just reported by NCI:

- Only a subset of CDEs are visible on the [Biospecimen Tab](https://htan-portal-nextjs-git-release-2-htan.vercel.app/explore?tab=biospecimen) (39 of 55, missing morphology code, tumor tissue types, etc.). Is it feasible to list all CDEs?

I think this is related to this ticket but we can file it separately if not

**Comment by onursumer** [comment]
@inodb The workaround proposed by PR #443 adds all children to every single schema object as `optionalDependencies`, so we end up having optional dependencies even for primitive types. For example `Number` ends up having `Integer` as an optional dependency. I think this is okay, but probably not ideal.

![number_optional_deps](https://user-images.githubusercontent.com/3604198/195391966-971afece-dbbd-462e-9c53-9e0052ddd113.png)

**Comment by milen-sage** [comment]
@inodb and @onursumer we will hand over the schema visualization data and code to you. We are waiting on these PRs to bw merges: https://github.com/ncihtan/data-models/pull/99, https://github.com/ncihtan/data-models/pull/94 since they fix issues with the schema that currently block generating visualization data.

I'm in touch with @adamjtaylor and @elv-sb on when the PRs will be approved.

**Comment by aclayton555** [comment]
@milen-sage Can you confirm if any further activity is pending here from the FAIR side? FYI @abbywall

**Comment by milen-sage** [comment]
The PRs^ have been merged. @linglp has generated the schema visualization and related schema data. @inodb we can meet during HTAN F2F to see how the schema data and visualization JS code can be plugged in the data portal? This could be a separate issue. It sounds like we could close this one?
